functions/source/soci-index-generator-lambda/utils/registry/registry.go

```diff
@@ -143,24 +158,66 @@ func (registry *Registry) GetManifest(ctx context.Context, repositoryName string
 	return manifest, nil
 }
 
-// Validate if a digest is a valid image manifest
-func (registry *Registry) ValidateImageManifest(ctx context.Context, repositoryName string, digest string) error {
-	manifest, err := registry.GetManifest(ctx, repositoryName, digest)
+func (registry *Registry) validateImageManifest(ctx context.Context, repositoryName string, digest string) error {
+	// Get the manifest content
+	imageRef, err := registry.GetManifest(ctx, repositoryName, digest)
+	if err != nil {
+		return err
+	}
+
+	// Valid image manifests must have a config with a valid media type
+	if imageRef.Config.MediaType == "" {
+		return fmt.Errorf("not a valid image manifest: empty config media type")
+	}
+
+	if !slices.Contains(ImageConfigMediaTypes, imageRef.Config.MediaType) {
+		return fmt.Errorf("not a valid image manifest: unexpected config media type: %s, expected one of: %v",
+			imageRef.Config.MediaType, ImageConfigMediaTypes)
+	}
+
+	return nil
+}
+
+func (registry *Registry) validateImageIndex(ctx context.Context, repositoryName string, digest string) error {
+	// Get the descriptor to check media type
+	descriptor, err := registry.HeadManifest(ctx, repositoryName, digest)
 	if err != nil {
 		return err
 	}
 
-	if manifest.Config.MediaType == "" {
-		return fmt.Errorf("Empty config media type.")
+	// Check if it's an image index by media type
+	if !images.IsIndexType(descriptor.MediaType) {
+		return fmt.Errorf("not a valid image index: unexpected media type: %s", descriptor.MediaType)
 	}
 
-	for _, configMediaType := range ImageConfigMediaTypes {
-		if manifest.Config.MediaType == configMediaType {
+	return nil
+}
+
+// ValidateImageDigest validates if a digest is valid based on SOCI index version requirements
+// For SOCI V1, only image manifests are supported
+// For SOCI V2, both image manifests and image indexes are supported
+func (registry *Registry) ValidateImageDigest(ctx context.Context, repositoryName string, digest string, sociIndexVersion string) error {
+	var err error
+	if sociIndexVersion == "V1" {
+		err = registry.validateImageManifest(ctx, repositoryName, digest)
+		if err != nil {
+			return err
+		}
+		log.Info(ctx, "Validated image manifest")
+	}
+	if sociIndexVersion == "V2" {
+		err := registry.validateImageIndex(ctx, repositoryName, digest)
```

Comment:
```suggestion
		err = registry.validateImageIndex(ctx, repositoryName, digest)
```

If both fail, this will always return nil as written.